charts/consul/test/acceptance/tests/partitions/partitions_test.go

```diff
@@ -0,0 +1,148 @@
+	}
+
+	if !cfg.UseKind {
+		t.Skipf("skipping this test because Admin Partition tests are only supported in Kind for now")
+	}
+
+	primaryContext := env.DefaultContext(t)
+	secondaryContext := env.Context(t, environment.SecondaryContextName)
+
+	ctx := context.Background()
+
+	primaryHelmValues := map[string]string{
+		"global.datacenter": "dc1",
+		"global.image":      "hashicorp/consul-enterprise:1.11.0-ent-alpha",
+
+		"global.adminPartitions.enabled": "true",
+		"global.enableConsulNamespaces":  "true",
+		"global.tls.enabled":             "true",
+
+		"server.exposeGossipAndRPCPorts": "true",
+
+		"connectInject.enabled": "true",
+		"controller.enabled":    "true",
+	}
+
+	if cfg.UseKind {
+		primaryHelmValues["global.adminPartitions.service.type"] = "NodePort"
+		primaryHelmValues["global.adminPartitions.service.nodePort.https"] = "30000"
+	}
+
+	releaseName := helpers.RandomName()
+
+	// Install the primary consul cluster in the default kubernetes context
+	primaryConsulCluster := consul.NewHelmCluster(t, primaryHelmValues, primaryContext, cfg, releaseName)
+	primaryConsulCluster.Create(t)
+
+	// Get the TLS CA certificate and key secret from the primary cluster and apply it to secondary cluster
+	tlsCert := fmt.Sprintf("%s-consul-ca-cert", releaseName)
+	logger.Logf(t, "retrieving ca cert secret %s from the primary cluster and applying to the secondary", tlsCert)
+	caCertSecret, err := primaryContext.KubernetesClient(t).CoreV1().Secrets(primaryContext.KubectlOptions(t).Namespace).Get(ctx, tlsCert, metav1.GetOptions{})
+	caCertSecret.ResourceVersion = ""
+	require.NoError(t, err)
+	_, err = secondaryContext.KubernetesClient(t).CoreV1().Secrets(secondaryContext.KubectlOptions(t).Namespace).Create(ctx, caCertSecret, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	tlsKey := fmt.Sprintf("%s-consul-ca-key", releaseName)
+	logger.Logf(t, "retrieving ca key secret %s from the primary cluster and applying to the secondary", tlsKey)
+	caKeySecret, err := primaryContext.KubernetesClient(t).CoreV1().Secrets(primaryContext.KubectlOptions(t).Namespace).Get(ctx, tlsKey, metav1.GetOptions{})
+	caKeySecret.ResourceVersion = ""
+	require.NoError(t, err)
+	_, err = secondaryContext.KubernetesClient(t).CoreV1().Secrets(secondaryContext.KubectlOptions(t).Namespace).Create(ctx, caKeySecret, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	var partitionSvcIP string
+	if !cfg.UseKind {
+		// Get the IP of the partition service to configure the external server address in the values file for the workload cluster.
+		partitionServiceName := fmt.Sprintf("%s-partition-secret", releaseName)
+		logger.Logf(t, "retrieving partition service to determine external IP for servers")
+		partitionsSvc, err := primaryContext.KubernetesClient(t).CoreV1().Services(primaryContext.KubectlOptions(t).Namespace).Get(ctx, partitionServiceName, metav1.GetOptions{})
+		require.NoError(t, err)
+		partitionSvcIP = partitionsSvc.Status.LoadBalancer.Ingress[0].IP
```

Comment:
Why do we have this logic if the tests only run on kind? I'm wondering if we should leave it out until it's needed, but maybe you wanted to keep it because we'll most likely need it in the future.

Reply:
Yeah. I added it for exactly that reason. We will need it in the long run.

Reply:
sounds good